src/languages/clojure.js

```diff
@@ -45,15 +45,15 @@ export default function(hljs) {
       'lazy-seq spread list* str find-keyword keyword symbol gensym force rationalize'
   };
 
-  const SIMPLE_NUMBER_RE = '[-+]?\\d+(\\.\\d+)?';
+  const NUMBER_RE = /[-+]?(((0[xX][0-9a-fA-F]+|0[0-7]+|[1-9][0-9]?[rR][0-9a-zA-Z]+)N?)|[0-9]+(\/[0-9]+|N|((\.[0-9]*)?([eE][+-]?[0-9]+)?M?)))/
```

Comment:
Please break this up into separate regex using `variants` as seen in other grammars, like `javascript`, etc...

Reply:
I initially did, but I couldn't get an order where everything matched like it should. 
Maybe negative lookaheads resolve that problem, though. 
I will look at this again.

Reply:
The order should be the same as they appear inside the larger regex as the variants will eventually get recompiled that way anyways... but for maintenance/readability it's much simpler to split these out.